src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -793,6 +794,16 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 TextSpan,
                 "BCP138",
                 "Loops are not currently supported.");
+       
+            public ErrorDiagnostic InvalidResourceTypePart(string typePart) => new(
+                TextSpan,
+                "BCP139",
+                $"The resource type part \"{typePart}\" is invalid. Nested resources must specify a single type, and optionally can specify a version using the format \"<type>@<apiVersion>\".");

```

Comment:
I've most commonly seen these referred to as a "type segment" rather than "type part". It would also be good to change "version" -> "api version"

Reply:
OK